internal/cmdrender/runtime/container.go

```diff
@@ -66,12 +70,28 @@ func (f *ContainerFn) Run(reader io.Reader, writer io.Writer) error {
 	cmd.Stdout = writer
 	cmd.Stderr = &errSink
 
-	pr.PkgPrintf(f.Path, "running function %q: ", f.Image)
+	pr.IndentPrintf(fnIdentation, "[RUNNING] %q\n", f.Image)
 	if err := cmd.Run(); err != nil {
-		pr.Printf("FAILED\n")
-		return fmt.Errorf("%s", errSink.String())
+		pr.IndentPrintf(fnIdentation, "[FAIL] %q\n", f.Image)
+		var exitCode int
+		if exitError, ok := err.(*exec.ExitError); ok {
+			exitCode = exitError.ExitCode()
+		} else {
+			return fmt.Errorf("cannot get function exit code: %w", err)
+		}
+		ff := fnoutput.FnFailure{
+			ExitCode: exitCode,
+			Stderr:   errSink.String(),
+		}
+		errStr, err := ff.String(f.TruncateOutput)
+		if err != nil {
+			return fmt.Errorf("failed to get function output string: %w", err)
+		}
+		pr.IndentPrintfE(fnFailureIdentation, errStr)
+		// TODO: write complete details to a file
+		return fmt.Errorf("function %q failed", f.Image)
```

Comment:
Note - error returned here will eventually be printed by the default error handler at the root level so that's going to pollute the output. We need a way to signal that to the default handler somehow. Will think more about that, @mortent has a PR which is related to this issue in general.